src/character.cpp

```diff
@@ -1976,6 +1976,14 @@ move_mode_id Character::current_movement_mode() const
     return move_mode;
 }
 
+int Character::fragile_terrain_weight_modifier() const
+{
+    if( !move_mode.is_valid() ) {
+        return 2; // defaut if move_mode is invalid
+    }
+    return move_mode->get_bash_weight_modifier();
```

Comment:
It is safe to assume the move mode is always valid. Also, instead of sprinkling the magic number `2` everywhere, you can have this chain to the `Creature::` function (another reason I think a multiplier is more clear - you need to multiply dividers to chain properly).